dev/devicelab/lib/framework/adb.dart

```diff
@@ -271,6 +285,63 @@ class AndroidDevice implements Device {
     };
   }
 
+  @override
+  Stream<String> get logcat {
+    final Completer<void> stdoutDone = new Completer<void>();
+    final Completer<void> stderrDone = new Completer<void>();
+    final Completer<void> processDone = new Completer<void>();
+    final Completer<void> abort = new Completer<void>();
+    bool aborted = false;
+    StreamController<String> stream;
+    stream = new StreamController<String>(
+      onListen: () async {
+        await adb(<String>['logcat', '--clear']);
+        final Process process = await startProcess(adbPath, <String>['-s', deviceId, 'logcat']);
```

Comment:
Seems pretty similar to the eval method in utils.dart, what do you think about extracting the common functionality? (e.g having a stream version of eval and use it in the Future<String> version)

Reply:
Yeah, I tried, but couldn't really get it to work cleanly. The `Process` API just isn't very conducive to terse code unfortunately.